xivo_bus/mixins.py

```diff
@@ -233,49 +250,82 @@ def __on_message_received(self, body, message):
         finally:
             message.ack()
 
+    def on_connection_error(self, exc, interval):
+        self.log.error(
+            'Broker connection error: %s, trying to reconnect in %s seconds...',
+            exc,
+            interval,
+        )
+        if self.should_stop:
+            raise SystemExit
+
+    def create_connection(self):
+        self.connection = self.__connection.clone()
+        return self.connection
+
+    @property
+    def is_running(self):
+        try:
+            is_running = self.connection.connected
+        except AttributeError:
+            is_running = False
+        return super(ConsumerMixin, self).is_running and is_running
+
+    @property
+    def should_stop(self):
+        return getattr(self, 'is_stopping', True)
+
     def on_consume_ready(self, connection, channel, consumers, **kwargs):
         if 'ready_flag' in kwargs:
-            kwargs['ready_flag'].set()
+            ready_flag = kwargs.pop('ready_flag')
+            ready_flag.set()
```

Comment:
Why we now need to pop the kwargs?

Reply:
This flag is only useful during initialization to unblock the main thread, this flag serves no other purpose once started and can be removed